dpnp/backend/kernels/dpnp_krnl_mathematical.cpp

```diff
@@ -181,47 +182,78 @@ void dpnp_floor_divide_c(void* result_out,
                          const size_t input2_shape_ndim,
                          const size_t* where)
 {
-    (void)input1_shape;
-    (void)input1_shape_ndim;
-    (void)input2_size;
-    (void)input2_shape;
-    (void)input2_shape_ndim;
     (void)where;
 
-    cl::sycl::event event;
-    _DataType_input1* input1 = reinterpret_cast<_DataType_input1*>(const_cast<void*>(input1_in));
-    _DataType_input2* input2 = reinterpret_cast<_DataType_input2*>(const_cast<void*>(input2_in));
+    if (!input1_size || !input2_size)
+    {
+        return;
+    }
+
+    _DataType_input1* input1_data = reinterpret_cast<_DataType_input1*>(const_cast<void*>(input1_in));
+    _DataType_input2* input2_data = reinterpret_cast<_DataType_input2*>(const_cast<void*>(input2_in));
     _DataType_output* result = reinterpret_cast<_DataType_output*>(result_out);
 
-    if constexpr ((std::is_same<_DataType_input1, double>::value || std::is_same<_DataType_input1, float>::value) &&
-                  std::is_same<_DataType_input2, _DataType_input1>::value)
+    std::vector<size_t> result_shape = get_result_shape(input1_shape, input1_shape_ndim,
+                                                            input2_shape, input2_shape_ndim); 
+
+    DPNPC_id<_DataType_input1>* input1_it;
+    const size_t input1_it_size_in_bytes = sizeof(DPNPC_id<_DataType_input1>);
+    input1_it = reinterpret_cast<DPNPC_id<_DataType_input1>*>(dpnp_memory_alloc_c(input1_it_size_in_bytes));
+    new (input1_it) DPNPC_id<_DataType_input1>(input1_data, input1_shape, input1_shape_ndim);
```

Comment:
I wouldn't recommend this type of initialization

Reply:
We can try copying input object from host to device instead of placement new.
```c++
DPNPC_id<_DataType_input1> input1(input1_data, input1_shape, input1_shape_ndim);
input1.broadcast_to_shape(result_shape);
const DPNPC_id<_DataType_input1>* input1_it = &input1;

DPNPC_id<_DataType_input1>* input1_it_device;
const size_t input1_it_size_in_bytes = sizeof(DPNPC_id<_DataType_input1>);
input1_it_device = reinterpret_cast<DPNPC_id<_DataType_input1>*>(dpnp_memory_alloc_c(input1_it_size_in_bytes));
auto copy_input1_it_event = DPNP_QUEUE.memcpy(input1_it_device, input1_it, input1_it_size_in_bytes);
copy_input1_it_event.wait();
```

Reply:
Yes we can copy but I would strong do not recommend it.
SYCL take care about this.